src/main/app/forms/validation/validators/checkCountry.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  registerDecorator,
+  ValidationArguments,
+  ValidationOptions,
+  ValidatorConstraint,
+  ValidatorConstraintInterface
+} from 'class-validator'
+import { Country } from 'app/common/country'
+
+import { PostcodeInfoClient, PostcodeInfoResponse } from '@hmcts/postcodeinfo-client'
+import * as config from 'config'
+import { request } from 'client/request'
+import { ValidationErrors as AddressValidationErrors } from 'app/forms/models/address'
+
+const postcodeClient = new PostcodeInfoClient(config.get<string>('postcodeLookup.apiKey'), request)
+
+@ValidatorConstraint()
+export class CheckCountryConstraint implements ValidatorConstraintInterface {
+
+  async validate (value: any | string, args?: ValidationArguments): Promise<boolean> {
+    if (value === undefined || value === null || value === '') {
+      return false
+    }
+    const postcodeInfoResponse: PostcodeInfoResponse = await postcodeClient.lookupPostcode(value)
+    if (!postcodeInfoResponse.valid) {
+      return true
+    }
+    const country = postcodeInfoResponse.country.name
+    const countries: Country[] = args.constraints[0]
+
+    return countries.filter(result => result.name.toLowerCase() === country.toLowerCase()).length > 0
+  }
+
+  defaultMessage (args: ValidationArguments) {
```

Comment:
Please simplify by providing generic error message saying `Country is not permitted` or similar. Current implementation strongly tights the default message that should be generic with specific validation rules for claimant / defendant country. That is not desirable. When default message is inappropriate then it can be overridden.